### DISCUSSION TITLE
Implement Google's Zanzibar-based authorization system

### DISCUSSION THREAD

**Discussion by juanjoDiaz** [keycloak/keycloak#19554] [Ideas]
Currently, Keycloak supports multiple authorization systems.

The new cool kid around the block are system based on Google's Zanzibar paper, a Relationship-based access control (ReBAC) approach. You can read more about it here: https://zanzibar.academy/

There are already a few projects in this space:
* [OpenFGA](https://openfga.dev/) by Auth0
* [Ora KEto](https://www.ory.sh/docs/keto) which used to be based on OPA policies and changed to Zanzibar
* [Permify](https://permify.co/)
* [Authzed](https://authzed.com/)

It would be very interesting to see how this model would fit in Keycloak authorization services.

**Comment by aaguiarz** [comment]
Hi @juanjoDiaz !

There's an integration with OpenFGA here https://www.infoq.com/news/2023/05/open-fine-grained-authorization/

Given both Keycloak and OpenFGA are CNCF-hosted projects now, it would be an interesting collaboration project :). 

If someone from Keycloak in interested in pursuing it please ping me.

**Comment by christian-2** [comment]
#21497 related in that it also mentions Zanzibar
